protos/core/Btfs.proto

```diff
@@ -0,0 +1,230 @@
+	bytes host_signature = 19; //host’s signature from the sub contract
+	bytes escrow_signature = 20； //escrow's signature 
+	enum State{
+		DRAFT = 0; //the contract was draft, but the escrow service did not received the money
+		SIGNED = 1;  // the escrow service received the money from renter, the contract was start.
+		UPLOADED = 2; //the first challenge was done, which means the shard was in the host, all pending payouts can be issued now
+		LOST = 3; // the challenge failed, the following payout should be hold or even stopped
+		CANCELED = 4; //if the shard was rebuild after LOST, or after some days of NOQUESTION, it go to CANCELED status
+		CLOSED = 5; // the all work in such contract was done
+	}
+	State state = 21;
+	bytes change_log = 22; //TBD one field to store the contract change log, need to discuss whether need the field
+	Timestamp last_modify_time = 23;
+	bytes guard_signature = 24;
+}
+
+
+message FileStoreMeta{
+	bytes renter_address = 1; // buyer wallet address
+	bytes file_hash = 2; // file’s hash
+	int64 file_size = 3; // file’s size
+	Timestamp rent_start = 4; //rent start time
+	Timestamp rent_end = 5; //rent end time
+	int check_frequence = 6; // shard check frequence in a year,  1 means check once per year
+	int64 guard_fee= 7; // reserve: the fee which guard will charge for supporting 
+	int64 escrow_fee = 8; //reserve: the fee which escrow will charge for supporting
+	int shard_count = 9; //shard's count, by default is 30
+	int minimum_shards = 10; //minimum count of shards which can recover the whole file, by default is 10
+	int recover_threshold = 11; //reserved field, indicate the threshold for starting to recover
+	repeated Contract Contracts = 12;
+	bytes escrow_address = 13;
+	bytes guard_address = 14;
+	bytes renter_signature = 15; //renter's signature for above information
+	Timestamp guard_receive_time = 16; //guard received time, which used to calculate the timeout for cancel
+	bytes change_log = 17; //TBD one field to store the contract change log, need to discuss whether need the field
+	Timestamp last_modify_time = 18;
+	bytes guard_signature = 19;
+	//TBD need to add one more field for status? 
+}
+
+message ChanllengeQuestion{
+	bytes shard_hash = 1 //shard hash 
+	bytes host_address = 2 //host's address
+	int chunk_index = 3 //the chunk's index for calculation
+	int random_nounce = 4 //the challenge questions's random nounce 
+	bytes expect_answer = 5 //the challenge expected answer
+}
+
+message ProofOfReplicateChallenge{
+	ChanllengeQuestion question = 1; //of course expect_answer should be clear before send to the host`
+	Timestamp challenge_time = 2 //challenge time from guard
+	bytes guard_address = 3 //guar's address
+	bytes guard_signature = 4 //guard signed for the challenge
+	bytes host_answer = 5 //challenge answer
+	Timestamp host_sign_time = 6 //response time from host
+	bytes host_signature = 7
+}
+
+
+message ChallengeQuestions{
+	bytes file_hash = 1; //file's hash
+	bytes shard_hash = 2;
```

Comment:
challenge questions per file per shard? so you need to store 30 challengequestions per *FileStoreMeta*? Isn't it easier to include in the struct directly?

Reply:
Add new message FileChallengeQuestions to group all shard questions.